src/lib/utils/Statistic/interpolate-data.test.js

```diff
@@ -0,0 +1,85 @@
+import { describe, expect, it } from 'vitest';
+import interpolateData from '$lib/utils/Statistic/interpolate-data';
+import parseInterval from '$lib/utils/intervals';
+
+describe('Interpolate dataset', () => {
+	it('returns interpolated data from 30 minutes interval to 5 minutes interval', () => {
+		const minInterval = parseInterval('5m');
+		const data = [
+			// TODO: not sure why we require this to be here for interpolation to occur?
```

Comment:
And here - I wasn't clear why there needed to be another entry in the data array. Would be great to make the test explicit to explain further.

Reply:
it's because the function only performs the interpolation if it detects there is a difference in the intervals returned. Will add that comment to the `interpolate-data` function.

Reply:
@chienleng thanks :) Given the desired minInterval is passed in, the context I think I'm missing was around the relationship of the start/end date of those StatsData objects that match the desired minInterval and the corresponding filtering - ie around https://github.com/opennem/openelectricity/pull/43/files/bc0fbba86875f99da82bcbcf2881b4f07c6e1946#diff-fc1bb786b10f9d5da7716d320830a20042c0a7e5b60670ff93f3f9798ad74beeR56-R66. This only stood out while I was trying to reverse engineer - I haven't yet dug into the broader usage when I'm sure this would be clear, so no need to address as such :)